Marlin/src/lcd/extui/mks_ui/draw_keyboard.cpp

```diff
@@ -160,7 +160,7 @@ static void lv_kb_event_cb(lv_obj_t *kb, lv_event_t event) {
           goto_previous_ui();
           break;
         case GCodeCommand:
-          if (ret_ta_txt[0] && !queue.ring_buffer.full(3)) {
+          if (ret_ta_txt[0] && !queue.ring_buffer.full(BUFSIZE - 1)) {
```

Comment:
This would require the buffer to have no more than 1 command in it, but we probably want to allow for the buffer to have more things.

Meanwhile, there may be a fundamental misunderstanding from Makerbase about how `queue.inject` works. It doesn't put anything into the G-code buffer right away. It sets up commands that will be injected before the regular G-code buffer. So it doesn't matter here if the buffer is "full" or "close to full" or not! The commands will be run before the next command(s) in the queue.